### PR TITLE
IGNITE-19493 Sql. Change query execution flow

### PR DESCRIPTION
**Pull request by zstan**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 1]
```diff
@@ -425,6 +421,18 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
                 .thenCompose(sqlNode -> {
                     boolean rwOp = dataModificationOp(sqlNode);
 
+                    boolean implicitTxRequired = outerTx == null;
+
+                    tx.set(implicitTxRequired ? txManager.begin(!rwOp) : outerTx);
+
+                    SchemaPlus schema = sqlSchemaManager.schema(schemaName);
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:428]:
```suggestion
                    SchemaPlus schema = sqlSchemaManager.activeSchema(schemaName, tx.startTimestamp().longValue());
```

Comment (AMashenkov) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:428]: SqlSchemaManager methods without version or timestamp must not be used, and it is better to drop them at all.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 2]
```diff
@@ -425,6 +421,18 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
...
+
+                    if (schema == null) {
+                        return CompletableFuture.failedFuture(
+                                new IgniteInternalException(SCHEMA_NOT_FOUND_ERR,
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:432]: Let's introduce a top-level class SchemaNotFoundException in public API.

File: modules/runner/src/integrationTest/sql/types/timestamp/test_incorrect_timestamp.test [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # name: test/sql/types/timestamp/test_incorrect_timestamp.test
 # description: Test out of range/incorrect timestamp formats
 # group: [timestamp]
-# Ignore https://issues.apache.org/jira/browse/IGNITE-15623
```
[the remark this answers is not in the document]
Comment (zstan) [reply modules/runner/src/integrationTest/sql/types/timestamp/test_incorrect_timestamp.test:4]: yep, but can it be here ? issue already fixed but mention still exist

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java (hunk 2 above)
Comment (zstan) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java]: done

**Review by AMashenkov — APPROVED** [review]

**Pull request closed without merging**